src/query/phrase_query/phrase_weight.rs

```diff
@@ -88,7 +143,7 @@ impl PhraseWeight {
 
 impl Weight for PhraseWeight {
     fn scorer(&self, reader: &SegmentReader, boost: Score) -> crate::Result<Box<dyn Scorer>> {
```

Comment:
why not having only this `scorer` function and get rid of `phrase_scorer_as_scorer`?

```rust
fn scorer(&self, reader: &SegmentReader, boost: Score) -> crate::Result<Box<dyn Scorer>> {
    match self.get_scorer_params(reader, boost)? {
        Some((term_postings_list, similarity_weight, fieldnorm_reader)) => {
            if self.slop == 0 {
                Ok(Box::new(ExactPhraseScorer::new(
                    term_postings_list,
                    similarity_weight,
                    fieldnorm_reader,
                    self.scoring_enabled,
                )))
            } else {
                Ok(Box::new(SlopPhraseScorer::new(
                    term_postings_list,
                    similarity_weight,
                    fieldnorm_reader,
                    self.scoring_enabled,
                    self.slop,
                )))
            }
        }
        _ => Ok(Box::new(EmptyScorer)),
    }
}
```